nautobot/docs/development/apps/api/platform-features/table-extensions.md

```diff
@@ -0,0 +1,64 @@
+# Extending Tables
+
++++ 2.9.99-alpha
+
+Apps can extend any model-based `Table` classes that are provided by the Nautobot core.
+
+The requirements to extend a table are:
+
+* The file must be named `table_extensions.py`
+* The variable `table_extensions` must be declared in that file, and contain a list of `TableExtension` subclasses
+* The `model` attribute of each `TableExtension` subclass must be set to a valid model name in the dotted pair format (`{app_label}.{model}`, e.g. `tenant.tenant` or `dcim.device`)
```

Comment:
For now, let's just focus on the models, and thus the list views. I'd be concerned with tightly coupling this to specific tables, and issues with this breaking as updates occur. Perhaps it can be implemented as an advanced option in the future, but not needed now.